torch/csrc/jit/passes/peephole_non_tensor.cpp

```diff
@@ -192,25 +193,49 @@ struct PeepholeOptimizeNonTensorImpl {
           node->output()->replaceAllUsesWith(node->input());
           changed = true;
         }
+      } else if (node->kind() == aten::Int) {
+        if (node->input()->type()->cast<IntType>()) {
+          GRAPH_UPDATE(
+              "Removing ", getHeader(node), " as input is already an integer");
+          node->output()->replaceAllUsesWith(node->input());
+          changed = true;
+        }
```

Comment:
Why is this not handled by the code right above it?

Reply:
aten::Int is converting scalar/float/bool to int. so shouldnt be handled by unchecked_cast